### ISSUE TITLE
I want the ability to rename a file name in a zip file

### ISSUE THREAD

**Issue by l54808821** [srikanth-lingala/zip4j#72]
I want the ability to rename a file name in a zip file
If I use the following code, I will corrupt the original file

        ZipParameters zipParameters = new ZipParameters();
        zipParameters.setFileNameInZip("jmeterTemplate/data/result1.csv");
        zipParameters.setSourceExternalStream(true);

        ZipFile zipFile = new ZipFile("D:\\aaaa\\jmeterTemplate.zip");
        zipFile.setFileNameCharset("GBK");
        FileHeader fileHeader1 = zipFile.getFileHeader("jmeterTemplate/data/result.csv");
        ZipInputStream inputStream = zipFile.getInputStream(fileHeader1);
        zipFile.addStream(inputStream,zipParameters);

**Comment by LeeYoung624** [comment]
I guess you are using a old version of zip4j.
I tested with the latest code, and I think this could be a workaround for you:
```
    ZipParameters parameters = new ZipParameters();
    parameters.setFileNameInZip("newFileName");

    ZipFile resource = new ZipFile("/home/temp.zip");
    FileHeader fileHeader = resource.getFileHeader("fileToBeRenamed");
    InputStream is = resource.getInputStream(fileHeader);

    resource.addStream(is, parameters);
    resource.removeFile("fileToBeRenamed");
```
I tested on Ubuntu and it works well. But it will throw exception on Windows. Hope this helps you.

**Comment by l54808821** [comment]
public static void renameFile(String zipPathName, String oldName, String newName) throws Exception {
        ZipParameters zipParameters = new ZipParameters();
        zipParameters.setFileNameInZip(newName);
        zipParameters.setSourceExternalStream(true);

        ZipFile zipFile = new ZipFile(zipPathName);
        zipFile.setFileNameCharset("GBK");
        FileHeader fileHeader = zipFile.getFileHeader(oldName);
        if (fileHeader != null) {
            ZipInputStream inputStream = zipFile.getInputStream(fileHeader);
            InputStream bufferedInputStream = IOUtils.toBufferedInputStream(inputStream);
            zipFile.addStream(bufferedInputStream, zipParameters);
            inputStream.close();
            bufferedInputStream.close();
            removeFile(zipFile,oldName);
        }
    }

**Comment by srikanth-lingala** [comment]
This is a good idea for a new feature in zip4j. But it is slightly complicated to implement. I could not integrate in into v2.2.1 which was released today. But I will include this feature in the next release of zip4j.

**Comment by LeeYoung624** [comment]
I'm working on this feature now. I think I can push the PR today.

**Comment by srikanth-lingala** [comment]
Feature added to zip4j. It is now possible to rename file(s) in a zip file. Please look at the documentation in README under the section "Rename files" for more info.

**Comment by srikanth-lingala** [comment]
Feature included in v2.4.0 released today
